website/pages/en/index.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+
+const CompLibrary = require('../../core/CompLibrary.js');
+const MarkdownBlock = CompLibrary.MarkdownBlock; /* Used to read markdown */
+const Container = CompLibrary.Container;
+const GridBlock = CompLibrary.GridBlock;
+
+const siteConfig = require(process.cwd() + '/siteConfig.js');
+
+function imgUrl(img) {
+  return siteConfig.baseUrl + 'img/' + img;
+}
+
+function docUrl(doc, language) {
+  return siteConfig.baseUrl + 'docs/' + (language ? language + '/' : '') + doc;
+}
+
+function pageUrl(page, language) {
+  return siteConfig.baseUrl + (language ? language + '/' : '') + page;
+}
+
+class Button extends React.Component {
```

Comment:
how much work would it be to reuse the same setup for scalafmt/scalameta docs? Would we have to copy-paste these components or is there a smarter way to share code?

Reply:
Docusaurus provides some basic building blocks (https://docusaurus.io/docs/en/api-pages.html#provided-components), the rest we'll have to copy paste.